server/bin/gen_fake_report.py

```diff
@@ -56,22 +96,47 @@ def _generate(self):
 
 
 def init_parser():
-    parser = argparse.ArgumentParser(description="Generate a fake report to be sent by virt-who")
+    parser = argparse.ArgumentParser(
+        description="Generate a fake report to be sent by virt-who"
+    )
     parser.add_argument('--hypervisors', metavar='N', type=int,
                         help="The number of hypervisors to include.",
                         action="store", dest="num_hypervisors")
-    parser.add_argument('--guests', metavar="N", type=int, dest="num_guests", default=1)
-    parser.add_argument('--reports', metavar="N", type=int, dest="num_reports", default=1)
-    parser.add_argument('--format', type=str, dest="format", choices=['async', 'sync'], help="The format of report to generate ('sync' or 'async')")
+    parser.add_argument('--guests', metavar="N", type=int, dest="num_guests",
+                        default=1)
+    parser.add_argument('--v', metavar="virtWhoType", type=str,
+                        help="Virt-who type (Default - libvirt)",
+                        dest="virt_type", default="libvirt")
+    parser.add_argument('--h', metavar="hypervisorType", type=str,
+                        help="Hypervisor type (Default - QEMU)",
+                        dest="hypervisor_type", default="QEMU")
+    parser.add_argument('--id', type=str, dest="hypervisor_id_format",
+                        choices=['uuid', 'hostname'], help="Hypervisor ID " +
+                        "format 'uuid' or 'hostname' (Default - uuid)",
+                        default="uuid")
+    parser.add_argument('--reports', metavar="N", type=int,
+                        dest="num_reports", default=1)
+    parser.add_argument('--format', type=str, dest="format",
+                        choices=['async', 'sync'], help="The format of " +
+                        "report to generate ('sync' or 'async')")
     parser.add_argument("-o", nargs="?", type=argparse.FileType('w'),
                         default=sys.stdout, dest='outfile')
     args = parser.parse_args()
 
     args.format = str.lower(args.format).strip()
+    args.hypervisor_id_format = str.lower(args.hypervisor_id_format).strip()
 
     return args
 
 
+def gen_random_string():
+    """
+    Generates random alphanumeric strings
+    """
+    return ''.join(random.choices(
```

Comment:
I'm getting the following:
```
[vagrant@candlepin vagrant]$ ./server/bin/gen_fake_report.py --hypervisors 1 --guests 2 --reports 1 --format async
Traceback (most recent call last):
  File "./server/bin/gen_fake_report.py", line 161, in <module>
    report = generator.generate()
  File "./server/bin/gen_fake_report.py", line 25, in generate
    return json.dumps(self._generate())
  File "./server/bin/gen_fake_report.py", line 94, in _generate
    range(0, self.hypervisors)]}
  File "./server/bin/gen_fake_report.py", line 79, in gen_hypervisor
    "name": "generated_hypervisor-" + gen_random_string(),
  File "./server/bin/gen_fake_report.py", line 136, in gen_random_string
    return ''.join(random.choices(
AttributeError: 'module' object has no attribute 'choices'
```
I think the method name is `choice`, not `choices`

Reply:
It looks like it's because I'm using python 2.7.5 (which is the default in Centos 7) and this method was only added in 3.6.
Let's make sure the script is compatible in general for both versions

Reply:
yes. It's because of Python 2.7, changes made to work on both python versions.